src/main/java/com/alkemy/ong/security/JwtTokenFilter.java

```diff
@@ -0,0 +1,54 @@
+package com.alkemy.ong.security;
+/*
+se ejecuta por cada petición y verifica el token en el caso concreto
+ */
+
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+public class JwtTokenFilter extends OncePerRequestFilter{
+
+    private final static Logger logger = LoggerFactory.getLogger(JwtTokenFilter.class);
+
+    @Autowired
+    JwtProvider jwtProvider;
+
+    @Autowired
+    UserDetailsServiceImpl userDetailsService;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest req, HttpServletResponse res, FilterChain filterChain) throws ServletException, IOException {
+        try{
+            String token= getToken(req);
+            if(token != null && jwtProvider.validateToken(token)){
+                String email = jwtProvider.getEmailFromToken(token);
+                UserDetails userDetails = userDetailsService.loadUserByUsername(email);
+                UsernamePasswordAuthenticationToken auth =
+                        new UsernamePasswordAuthenticationToken(userDetails, null , userDetails.getAuthorities() );
+                SecurityContextHolder.getContext().setAuthentication(auth);
+            }
+        }catch(Exception e){
+            logger.error("fail en el metodo doFilter" + e.toString());
+
+        }
+        filterChain.doFilter(req,res);
+    }
+    private String getToken(HttpServletRequest request){
+        String header= request.getHeader("Authorization");
+        if(header != null && header.startsWith("Bearer"))
```

Comment:
Falta un espacio despues del Bearer:
```suggestion
    }
    private String getToken(HttpServletRequest request){
        String header= request.getHeader("Authorization");
        if(header != null && header.startsWith("Bearer "))
```